hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/portable/FieldDefinitionImpl.java

```diff
@@ -113,35 +110,16 @@ public boolean equals(Object o) {
         if (o == null || getClass() != o.getClass()) {
             return false;
         }
-
         FieldDefinitionImpl that = (FieldDefinitionImpl) o;
-        if (index != that.index) {
-            return false;
-        }
-        if (factoryId != that.factoryId) {
-            return false;
-        }
-        if (classId != that.classId) {
-            return false;
-        }
-        if (version != that.version) {
-            return false;
-        }
-        if (fieldName != null ? !fieldName.equals(that.fieldName) : that.fieldName != null) {
-            return false;
-        }
-        return type == that.type;
+        return index == that.index
+                && Objects.equals(fieldName, that.fieldName)
+                && type == that.type
+                && portableId.equals(that.portableId);
     }
 
     @Override
     public int hashCode() {
-        int result = index;
-        result = 31 * result + (fieldName != null ? fieldName.hashCode() : 0);
-        result = 31 * result + (type != null ? type.hashCode() : 0);
-        result = 31 * result + factoryId;
-        result = 31 * result + classId;
-        result = 31 * result + version;
-        return result;
+        return Objects.hash(index, fieldName, type, portableId);
```

Comment:
hashCode will be different than before. Does it matter eg. for RU?
This one includes `factoryId` unlike in `ClassDefinitionImpl`

Reply:
note that `assertEquals(Objects.hash(1, 2, 3, 4), Objects.hash(1, 2, Objects.hash(3, 4)));` is not true

Reply:
This hash code is not used anywhere.